src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -1031,6 +1031,39 @@ TEST_CASE("refund account merged", "[tx][soroban][merge]")
     checkTx(1, r, txNO_ACCOUNT);
 }
 
+TEST_CASE("refund still happens on bad auth", "[tx][soroban]")
+{
+    SorobanTest test;
+
+    const int64_t startingBalance =
+        test.getApp().getLedgerManager().getLastMinBalance(50);
+
+    auto a1 = test.getRoot().create("A", startingBalance);
+    auto b1 = test.getRoot().create("B", startingBalance);
+    auto wasm = rust_bridge::get_test_wasm_add_i32();
+    auto resources = defaultUploadWasmResourcesWithoutFootprint(wasm);
+    auto tx =
+        makeSorobanWasmUploadTx(test.getApp(), a1, wasm, resources, 100000);
+
+    auto a1PreTxBalance = a1.getBalance();
+    auto setOptions = txtest::setOptions(setMasterWeight(0));
+    setOptions.sourceAccount.activate() = toMuxedAccount(a1);
+
+    auto classicSetOptionsTx = b1.tx({setOptions});
+    classicSetOptionsTx->addSignature(a1.getSecretKey());
+    std::vector<TransactionFrameBasePtr> txs = {classicSetOptionsTx, tx};
+    auto r = closeLedger(test.getApp(), txs);
+    checkTx(0, r, txSUCCESS);
+
+    // The source account of the soroban tx was merged during the classic phase
+    checkTx(1, r, txBAD_AUTH);
+
+    auto a1PostTxBalance = a1.getBalance();
+
+    // The difference would be much higher if the refund did not happen
```

Comment:
FWIW the difference without refund would be exactly `100100`.

Reply:
The difference I see without the refund is `1040871`. Why do you think it should be `100100`? Is there an issue here?

Reply:
Sorry for the confusion, this looks fine actually because `100000` is inclusion fee, not the total resource fee (which I was thinking about). Inclusion fee has nothing to do with refunds actually; can you change it to just `100`? .
The default additional refundable fee is `1'000'000` and then we use a rough computation to compute the necessary resource fee, so your number looks correct. You can just update the comment to say that the initial charge is around `DEFAULT_TEST_RESOURCE_FEE`, which is >> than 40K.

Reply:
Ah ok makes sense. Updated.